chunky/src/java/se/llbit/chunky/renderer/scene/SynchronousSceneManager.java

```diff
@@ -287,4 +291,48 @@ public void discardSceneChanges() {
     }
   }
 
+  /**
+   * Find and resolve the directory for a given scene name. If the scene is saved in the /scenes/ folder, it will return
+   * the scenes folder.
+   *
+   * If the scene is found in a folder inside the /scenes/ directory (eg. /scenes/some_scene/) that directory will be
+   * returned.
+   *
+   * Otherwise, a new directory in the /scenes/ folder will be created for the given scene and that said directory will
+   * be returned.
+   *
+   * @param sceneName The name of the scene to resolve the directory for.
+   * @return The directory holding the given scene
+   */
+  private File resolveSceneDirectory(String sceneName) {
+    File defaultDirectory = new File(PersistentSettings.getSceneDirectory(), sceneName);
+    //If this file doesnt exist, we assume we are making a new scene, loading an old scene from the scene directory,
```

Comment:
The comments here are _very_ verbose. It's not bad but it's not like this is a very complicated algorithm. A few short comments might also do it – a lot of comments like here actually make the code less readable.

Reply:
Understandable. I just wanted to make sure it was clear to people without them having to read through the rest of the whole program to figure out what exactly it does. Felt like I had to when I started working on chunky hah

Reply:
We should aim to have this level of comments on the path tracing code. That would really help :-)